### PR TITLE
Feature/hardhat flatten deploy verify tasks

### PR DESCRIPTION
**Pull request by majkic99**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,3 +38,11 @@ block number the fork starts from. If it starts from an old state some tests mig
 `npm run test [network] [test-file]` - will run a test to the specified network by calling the script from the `/test` folder
 
 `npm run verify [network] [contract-name]` - will verify contract based on address and arguments from `/deployments` folder
+
+## Custom hardhat tasks
+
+`npx hardhat customFlatten [file-path]` -  will flatten contract that is ready for deployment
```
Comment (yippee-ki-yay) [inline README.md:44]: And put it where?

File: hardhat.config.js [hunk 2]
```diff
@@ -54,3 +70,30 @@ module.exports = {
         forkNetwork: '1',
     },
 };
+
+// eslint-disable-next-line no-undef
+task('fladepver', 'Deploys and verifies contract on etherscan')
```
Comment (yippee-ki-yay) [inline hardhat.config.js:75]: This should be in separate file in script, not hardcoded in config

File: scripts/hardhat-tasks-functions.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    case 'rinkeby':
+        url = 'https://api-rinkeby.etherscan.io/api';
+        break;
+    case 'goerli':
+        url = 'https://api-goerli.etherscan.io/api';
+        break;
+    default:
+    }
+    const tx = await axios.post(url, params, config);
+    console.log(tx.data);
```
Comment (yippee-ki-yay) [inline scripts/hardhat-tasks-functions.js:101]: Link to etherscan

File: scripts/hardhat-tasks-functions.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        solVersion = 'v0.7.6+commit.7338295f';
+    }
+    params.append('compilerversion', solVersion);
+    params.append('optimizationUsed', hardhatSettings.solidity.settings.optimizer.enabled ? 1 : 0);
+    params.append('runs', hardhatSettings.solidity.settings.optimizer.runs);
+    /// @notice : MIT license
+    params.append('licenseType', 3);
+
+    let url;
+    switch (network) {
```
Comment (yippee-ki-yay) [inline scripts/hardhat-tasks-functions.js:82]: `
let url = 'https://api.etherscan.io/api';

if (network !== 'mainnet') {
  url = `https://api-${network}.etherscan.io/api`;
}

`

Comment (majkic99) [reply scripts/hardhat-tasks-functions.js:82]: good thinking

File: scripts/hardhat-tasks-functions.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+function sleep(ms) {
+    return new Promise((resolve) => {
+        setTimeout(resolve, ms);
+    });
+}
+async function deployContract(contractName, gasPriceSelected) {
+    await execShellCommand('npx hardhat compile');
+    const overrides = {
+        // The price (in wei) per unit of gas
+        gasPrice: hre.ethers.utils.parseUnits(gasPriceSelected, 'gwei'),
```
Comment (yippee-ki-yay) [inline scripts/hardhat-tasks-functions.js:29]: Sanity check for that number

File: scripts/hardhat-tasks-functions.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    };
+    const [deployer] = await hre.ethers.getSigners();
+    console.log('Deploying contracts with the account:', deployer.address);
+
+    console.log('Account balance:', (await deployer.getBalance()).toString());
+    const Contract = await hre.ethers.getContractFactory(
+        `contracts/flattened/${contractName}.sol:${contractName}`,
+    );
+    const contract = await Contract.deploy(overrides);
```
Comment (yippee-ki-yay) [inline scripts/hardhat-tasks-functions.js:38]: Can we add a confirm yes/no prompt and info (what we are deploying, what network, how much it will cost)

**Review by yippee-ki-yay — CHANGES_REQUESTED** [review]
Script works but some small improvements needed

**Comment by yippee-ki-yay** [comment]
The tools worked correctly on mainnet, after a few minor fixes. One last thing is to add an optional 'nonce' parameter so if the deployment tx gets stuck we can manually override it by using the same deployment nonce.

**Comment by majkic99** [comment]
Added optional nonce parameter for overriding stuck transactions

**Review by yippee-ki-yay — APPROVED** [review]

**Pull request merged**